src/openassetio-core/include/openassetio/BatchElementError.hpp

```diff
@@ -45,9 +45,8 @@ class BatchElementError final {
     kUnknown = OPENASSETIO_BatchErrorCode_kUnknown,
     /**
      * Error code used during @ref entity resolution if a supplied
-     * entity reference does not have a meaningful string
-     * representation, or it is a valid reference format that doesn't
-     * exist.
+     * entity reference is valid but the entity does not exist or
+     * does not have any meaningful representation.
```

Comment:
[question] I updated the docs here from the ones copy-pasted from `exceptions.py` - prompted by "have a meaningful string representation", which I assume is from the dark ages before traits were invented. Is this OK?

Reply:
Good spot! Yes that is definitely out of date. As we say that missing traits data should result in the trait not being set, then it would most likely not result in an EntityResolutionError, how about something like this?

```
Error code used during @ref resolve "entity resolution" when the reference
itself is valid, but it is not possible to retrieve data for the referenced 
@ref entity. This could be because it does not exist, or a read-only entity
is being resolved for write. This should not be used if an entity does not
have a requested trait - simply do not set that trait in the resulting data.
```

Reply:
Updated to mostly match your suggestion in 5578135b